libethereum/DatabasePaths.h

```diff
@@ -0,0 +1,39 @@
+// Aleth: Ethereum C++ client, tools and libraries.
+// Copyright 2014-2019 Aleth Authors.
+// Licensed under the GNU General Public License, Version 3.
+
+#pragma once
+
+#include "libdevcore/FixedHash.h"
+#include <boost/filesystem/path.hpp>
+
+namespace dev
+{
+namespace eth
+{
+class DatabasePaths
+{
+public:
+    DatabasePaths(boost::filesystem::path const& _rootPath, h256 const& _genesisHash) noexcept;
+    DatabasePaths() = default;
+    void setPaths(boost::filesystem::path const& _rootPath, h256 const& _genesisHash) noexcept;
```

Comment:
@gumb0 I like the immutable idea, I went with the current approach since I was hesitant to use dynamic allocation for something trivial like paths. I can use unique_ptr and remove the default ctor + setter method which will effectively make the class immutable.